src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableStore.kt

```diff
@@ -128,6 +128,23 @@ class VariableStore(
             ?.let { fetchVariableOrNull(it) }
       }
 
+  fun fetchListByStableIds(stableIds: List<StableId>): List<Variable> =
+      with(VARIABLES) {
+        dslContext
+            .select(ID)
+            .from(VARIABLES)
+            .where(STABLE_ID.`in`(stableIds))
+            .and(
+                ID.notIn(
+                    dslContext
+                        .select(REPLACES_VARIABLE_ID)
+                        .from(VARIABLES)
+                        .where(REPLACES_VARIABLE_ID.isNotNull())))
+            .fetch(ID)
+            .filterNotNull()
```

Comment:
I assume the `filterNotNull` is because `ID` is declared as nullable, even though in reality it can never be null in this query. That is, it's just to make the Kotlin type system happy. If so, the slightly more efficient way to do this would be to do `.fetch(ID.asNonNullable())`. The `asNonNullable` extension function is a runtime no-op that does a type cast to remove nullability from a jOOQ field.